src/input.rs

```diff
@@ -258,7 +258,11 @@ pub(crate) struct InputReader<'a> {
 impl<'a> InputReader<'a> {
     fn new<R: BufRead + 'a>(mut reader: R) -> InputReader<'a> {
         let mut first_line = vec![];
-        reader.read_until(b'\n', &mut first_line).ok();
+        reader
+            .by_ref()
+            .take(1024)
```

Comment:
forgive my ignorance, but does this not mean that if a text file with a first line longer than 1024 chars is read, that first line would get truncated at 1024 chars?

Reply:
Only for the purposes of checking whether the file is binary. The check down below:
```
else if content_type != Some(ContentType::BINARY) && first_line.last() != Some(&b'\n') {
    reader.read_until(b'\n', &mut first_line).ok();
}
```
will complete the first line if the content is not binary. If the content is binary, bat wont show it anyway.

Actually, maybe I need to also check for `self.config.show_nonprintable`.